src/Aspire.Hosting/Publishing/ManifestPublisher.cs

```diff
@@ -11,11 +12,14 @@ namespace Aspire.Hosting.Publishing;
 
 internal class ManifestPublisher(ILogger<ManifestPublisher> logger,
                                IOptions<PublishingOptions> options,
-                               IHostApplicationLifetime lifetime) : IDistributedApplicationPublisher
+                               IHostApplicationLifetime lifetime,
+                               IConfiguration configuration
+                               ) : IDistributedApplicationPublisher
 {
     private readonly ILogger<ManifestPublisher> _logger = logger;
     private readonly IOptions<PublishingOptions> _options = options;
     private readonly IHostApplicationLifetime _lifetime = lifetime;
+    private readonly IConfiguration _configuration = configuration;
```

Comment:
Passing through some configuration information so that we don't have to rely on Azure.Provisioning reading tenant, subscription and environment name from environment variables directly (makes testing easier).